build_runner/lib/src/generate/build_definition.dart

```diff
@@ -216,4 +219,15 @@ class BuildDefinitionLoader {
       }
     }
   }
+
+  Iterable<AssetId> _listGeneratedAssetIds() sync* {
+    var glob = new Glob('$generatedOutputDirectory/**');
+    for (var id in _reader.findAssets(glob)) {
+      var packagePath = id.path.substring(generatedOutputDirectory.length + 1);
```

Comment:
This can potentially cause issues with relative paths given where this is run. This path should always be relative to the root.

Reply:
Synced offline. I originally thought this was a problem but since we're going through `_reader.findAssets` it will default to relative to the root package so this will always work no matter our `cwd`.